internal/api/middleware.go

```diff
@@ -53,3 +54,54 @@ func (s *Server) AuthMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
 		return next(c)
 	}
 }
+
+// VaultAuthMiddleware verifies JWT tokens and ensures users can only access their own vaults
+func (s *Server) VaultAuthMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		// Get token from header
+		authHeader := c.Request().Header.Get("Authorization")
+		if authHeader == "" {
+			return c.JSON(http.StatusUnauthorized, map[string]string{
+				"error": "Missing authorization header",
+			})
+		}
+
+		// Extract token from Bearer format
+		tokenParts := strings.Split(authHeader, " ")
+		if len(tokenParts) != 2 || tokenParts[0] != "Bearer" {
+			return c.JSON(http.StatusUnauthorized, map[string]string{
+				"error": "Invalid authorization header format",
+			})
+		}
+
+		// Validate token and get claims
+		claims, err := s.authService.ValidateToken(tokenParts[1])
+		if err != nil {
+			return c.JSON(http.StatusUnauthorized, map[string]string{
+				"error": "Invalid or expired token: " + err.Error(),
```

Comment:
Let's just log the error , and don't bleed the error information to the client side